stm/stm.ml

```diff
@@ -370,6 +321,12 @@ end = struct (* {{{ *)
   open Printf
 
   let print_dag vcs () =
+
+    (* Due to threading, this will be called from the toplevel with
+       we_are_parsing set to true, as indeed, the toplevel is waiting
+       for input . What a race! XD *)
+    Flags.we_are_parsing := false;
```

Comment:
I'm not following.  This boolean is there to detect "commands" that access the system state (the global environment) while parsing.  I.e. commands that behave differently if I parse;parse;exec;exec or parse;exec;parse;exec.

Why are you flipping this bit here?